### PR TITLE
IB: change blocking verbs calls to non-blocking.

### PR DESCRIPTION
**Pull request by jiajuncao**

### CODE REVIEW

File: contrib/infiniband/ibvctx.c [hunk 1]
```diff
@@ -1104,9 +1106,34 @@ int _get_cq_event(struct ibv_comp_channel * channel,
...
   internal_channel = ibv_comp_to_internal(channel);
+  flags = fcntl(internal_channel->real_channel->fd, F_GETFL, NULL);
+
+  // We need to change the call to non-blocking mode
+  if (flags & O_NONBLOCK == 0) {
+    int ms_timeout = 100;
+    struct pollfd my_pollfd = {
+      .fd = internal_channel->real_channel->fd,
+      .events = POLLIN,
+      .revents = 0
```
Comment (gc00) [inline contrib/infiniband/ibvctx.c]: For consistency with the rest of DMTCP, could we use the older style:
```
struct pollfd my_pollfd;
my_pollfd.fd = internal_channel->real_channel->fd;
```
and so on.  (If others have opinions about starting to use the new syntax now, feel free to add your comments.)

File: contrib/infiniband/ibvctx.c [hunk 2]
```diff
@@ -1132,6 +1160,31 @@ int _get_async_event(struct ibv_context * ctx, struct ibv_async_event * event)
...
+  int flags = fcntl(internal_ctx->real_ctx->async_fd, F_GETFL, NULL);
+
+  // We need to change the call to non-blocking mode
+  if (flags & O_NONBLOCK == 0) {
+    int ms_timeout = 100;
+    struct pollfd my_pollfd = {
+      .fd = internal_ctx->real_ctx->async_fd,
+      .events = POLLIN,
+      .revents = 0
+    };
```
Comment (gc00) [inline contrib/infiniband/ibvctx.c]: Same comment as earlier about the newer syntax for initializing a struct.

**Comment by jiajuncao** [comment]
This `designated initializer` was introduced in C99. Personally I like this syntax, which makes the code easy to read. But I'm open to other opinions as well. @rohgarg, @karya0, what do you think?

**Comment by gc00** [comment]
@karya0, you were proposing a formal coding style for DMTCP-3.0.  You had found a good style.  Does that style say anything about initializers (good, bad, indifferent?).  Thanks.

**Comment by gc00** [comment]
@jiajuncao and @karya0, this PR seems to have become orphaned.  Jiajun, could you force another check of this PR just in case the problem with "Some checks haven't completed yet" is a fluke?  Also, please resolve the conflicts, and then let's finish reviewing this.  Thanks.

**Pull request merged**